contracts/token/ERC721/ERC721.sol

```diff
@@ -330,9 +330,28 @@ contract ERC721 is Context, ERC165, IERC721 {
         if (!to.isContract()) {
             return true;
         }
-
-        bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
-        return (retval == _ERC721_RECEIVED);
+        bytes memory payload = abi.encodeWithSelector(
+            IERC721Receiver(to).onERC721Received.selector,
+            msg.sender,
+            from,
+            tokenId,
+            _data
+        );
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory returndata) = to.call(payload);
```

Comment:
Can you inline the value of `payload` here?

Reply:
Do you mean to change the solidity to.call() function to inline assembly call like this one:  call(g, a, v, in, insize, out, outsize)?

Reply:
What I meant was writing `to.call(abi.encodeWithSelector(...)`. It's not that important but it's what we prefer.